runtime/client/client.go

```diff
@@ -61,9 +65,23 @@ func (o *Options) BindFlags(fs *pflag.FlagSet) {
 		"The maximum burst queries-per-second of requests sent to the Kubernetes API.")
 }
 
-// GetConfigOrDie wraps ctrl.GetConfigOrDie and sets the configured Options, returning the modified rest.Config.
+// GetConfigOrDie wraps ctrl.GetConfigOrDie and checks if the Kubernetes apiserver
+// has PriorityAndFairness flow control filter enabled. If true, it returns a rest.Config
+// with client side throttling disabled. Otherwise, it returns a modified rest.Config
+// configured with the provided Options.
 func GetConfigOrDie(opts Options) *rest.Config {
 	config := ctrl.GetConfigOrDie()
+	enabled, err := flowcontrol.IsEnabled(context.Background(), config)
+	if err != nil {
+		log.Fatalf("could not check if the server has PriorityAndFairness flow control filter enabled: %s", err)
```

Comment:
Is this intentionally a fatal log? If we can't determine flow control, shouldn't it continue with the existing behavior instead of ending the whole program?

Reply:
Yes it is. The function name indicates that it forces the program to exit if it isn't successful. As @hiddeco said in his above comment, it'd be wise to have a counterpart which errors out instead and have this logic there. Do you think the flowcontrol check should only be in the other function or does it make sense to have it in this function as well in it's current form?

Reply:
I don't think we should be introducing such fatal code when we have the control to handle it gracefully. In this case, as this is a new feature, if this is requested against an old API server which doesn't support this API, it'll receive an error and that'll cause the whole program to stop. Keeping with the current behavior, if there's no flow control support, just continue with the client side configurations.

Reply:
> Otherwise, it returns a modified rest.Config configured with the provided Options.

It also doesn't match with the description of the function.